packaging/test_plugin_unified_storage_tiering.py

```diff
@@ -787,18 +798,41 @@ def test_open_read_close_triggers_restage__issue_303(self):
                 invoke_storage_tiering_rule()
                 delay_assert_icommand(self.user1, f"ils -L {logical_path}", 'STDOUT', self.tier2)
 
-                # Now see if the object is restaged appropriately on open/read/close.
-                self.user1.assert_icommand(["irods_test_read_object", logical_path], "STDOUT")
+                # Now see if the object is restaged appropriately on access.
+                access_command(*access_command_args, **access_command_kwargs)
                 # Wait for the object to be trimmed before checking destination to avoid timing issues.
                 lib.delayAssert(
                     lambda: lib.replica_exists_on_resource(self.user1, logical_path, self.tier2) == False)
-                self.assertTrue(lib.replica_exists_on_resource(self.user1, logical_path, self.tier1))
+                self.assertTrue(lib.replica_exists_on_resource(self.user1, logical_path, expected_destination_resource))
 
             finally:
                 if os.path.exists(filename):
                     os.unlink(filename)
                 self.user1.assert_icommand(["irm", "-f", logical_path])
 
+    def test_open_read_close_triggers_restage__issue_303(self):
+        filename = "test_open_read_close_triggers_restage__issue_303"
+        logical_path = "/".join([self.user1.session_collection, filename])
+        self.accessing_existing_data_object_triggers_restage(
+            logical_path, self.tier1, self.user1.assert_icommand, ["irods_test_read_object", logical_path], "STDOUT")
+
+    def test_replica_open_read_replica_close_triggers_restage__issue_316(self):
+        filename = "test_replica_open_read_replica_close_triggers_restage__issue_316"
+        logical_path = "/".join([self.user1.session_collection, filename])
+        self.accessing_existing_data_object_triggers_restage(
+            logical_path, self.tier1, self.user1.assert_icommand, ["istream", "read", logical_path], "STDOUT")
+
+    def test_replica_open_write_replica_close_triggers_restage__issue_316(self):
+        filename = "test_replica_open_read_replica_close_triggers_restage__issue_316"
+        logical_path = "/".join([self.user1.session_collection, filename])
+        self.accessing_existing_data_object_triggers_restage(
+            logical_path,
+            self.tier1,
+            self.user1.assert_icommand,
+            ["istream", "read", logical_path],
```

Comment:
Test name says "write", but `istream read` is used?

Reply:
Ah, good find.

Reply:
Done. TIL: one must target an existing replica in order for `istream` to target that replica rather than creating a new one on the default resource...